### ISSUE TITLE
Impossible uninitialized constant (regression in Rubygems 2.2)

### ISSUE THREAD

**Issue by indirect** [rubygems/rubygems#784]
I have no idea how this is even possible, but every single Bundler build on Ruby 1.8.7 is failing right now because of Rubygems 2.2.0 raising `uninitialized constant Gem::UserInteraction (NameError)` when I try to run `ronn` after installing the gem.

I can't reproduce on my machine locally, and the backtrace points to `lib/rubygems/ext/builder.rb:12`, which is several lines after `require 'rubygems/user_interaction'`. I believe this is the same error causing Rubygems builds on Travis to fail against 1.8.7 as well.

Any ideas? Full backtrace [here](https://gist.github.com/indirect/90a19aa3eee72e412bfa).

**Comment by indirect** [comment]
Any chance this will be fixed soon, now that it has a reproduction?

**Comment by steakknife** [comment]
Confirmed that @voxik 's patch appears to resolve this issue on MRI 2.2.0 (e.g., encountered while running bundler's `rake spec`).

## Before patch
``` 
$ rake spec
groff -Wall -mtty-char -mandoc -Tascii lib/bundler/man/bundle-exec | col -b > lib/bundler/man/bundle-exec.txt
/usr/local/ruby/ruby-2.2.0/bin/ruby -S ronn --roff --pipe man/bundle-install.ronn > lib/bundler/man/bundle-install
/Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/lib/ronn/document.rb:191:in `sniff': uninitialized constant Ronn::Document::Markdown (NameError)
	from /Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/lib/ronn/document.rb:76:in `initialize'
	from /Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/bin/ronn:166:in `new'
	from /Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/bin/ronn:166:in `block in <top (required)>'
	from /Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/bin/ronn:166:in `map'
	from /Users/bmf/.gem/ruby/2.2.0/gems/ronn-0.7.3/bin/ronn:166:in `<top (required)>'
	from /Users/bmf/.gem/ruby/2.2.0/bin/ronn:23:in `load'
	from /Users/bmf/.gem/ruby/2.2.0/bin/ronn:23:in `<main>'
rake aborted!
Command failed with status (1): [/usr/local/ruby/ruby-2.2.0/bin/ruby -S ron...]
/Users/bmf/Projects/bundler/Rakefile:220:in `block (3 levels) in <top (required)>'
/Users/bmf/Projects/bundler/Rakefile:23:in `block in invoke'
/Users/bmf/Projects/bundler/Rakefile:22:in `invoke'
Tasks: TOP => spec => man:build => man:build_all_pages => lib/bundler/man/bundle-install.txt => lib/bundler/man/bundle-install
(See full trace by running task with --trace)
```

## After patch
```
$ rake spec
groff -Wall -mtty-char -mandoc -Tascii lib/bundler/man/bundle-install | col -b > lib/bundler/man/bundle-install.txt
/usr/local/ruby/ruby-2.2.0/bin/ruby -S ronn --roff --pipe man/bundle-package.ronn > lib/bundler/man/bundle-package
groff -Wall -mtty-char -mandoc -Tascii lib/bundler/man/bundle-package | col -b > lib/bundler/man/bundle-package.txt
/usr/local/ruby/ruby-2.2.0/bin/ruby -S ronn --roff --pipe man/bundle-platform.ronn > lib/bundler/man/bundle-platform
groff -Wall -mtty-char -mandoc -Tascii lib/bundler/man/bundle-platform | col -b > lib/bundler/man/bundle-platform.txt
...
```

**Comment by voxik** [comment]
@steakknife But that is just one part of the issue ... but the other part might be worth of moving into separate ticket anyway :)

**Comment by steakknife** [comment]
@voxik  True.  Probably best to just have separate PR's, and use this as the meta-ticket.  Smaller fixes > mega refactoring.  The submitted PR unblocks bundler w/ latest MRI, which seems important to release sooner.  (I'm not familiar enough with either codebase to be useful for an encompassing fix.)

Also, might want to consider whether order is important for other gems, e.g., have a separate method for path prepend vs append... because at scale, it could be an edge-case that matters (e.g., real-world apps).
